### DISCUSSION TITLE
Server-side code exposed on the client side with getServerSideProps?

### DISCUSSION THREAD

**Discussion by Egor-Koldasov** [vercel/next.js#25225] [Help] [answered]
https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering
From the docs, it looks like in getServerSideProps function you have full access to the server environment. This code basically runs on the server.
But wouldn't that code be also bundled to the client-side exposing all the code that is written to run on the server?

I look at the example here https://nextjs.org/docs/authentication#authenticating-server-rendered-pages
It suggests putting the server-side authorization code right to the page file.

**Comment by lkbr** [comment, marked as answer]
Next ensures code wrapped in `getServerSideProps` is not included in the client bundle. Take a look at this [note](https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly): 

> Note that getStaticProps runs only on the server-side. It will never be run on the client-side. It won’t even be included in the JS bundle for the browser. [...]

Which links to this [tool](https://next-code-elimination.vercel.app/) thats helps you verify your code isn't included in the client bundle.

**Reply by Egor-Koldasov** [reply]
Thanks a lot, that makes sense.
